apollo-backend/src/main/java/io/logz/apollo/kubernetes/KubernetesMonitor.java

```diff
@@ -146,6 +151,20 @@ public void monitor() {
         }
     }
 
+    private boolean isDeployedEnvironmentConcurrencyLimitPermitsDeployment(Deployment deployment) {
+        Integer concurrencyLimit = environmentDao.getEnvironment(deployment.getEnvironmentId()).getConcurrencyLimit();
+        if (concurrencyLimit != null && concurrencyLimit > 0) {
+            long runningDeploymentOnEnvironment = deploymentDao.getAllRunningDeployments()
+                    .stream()
+                    .filter(runningDeployment -> runningDeployment.getEnvironmentId() == deployment.getEnvironmentId())
+                    .count();
+
+            return runningDeploymentOnEnvironment < concurrencyLimit;
```

Comment:
Each deployment is a single one? or per topic?
What is the desired behavior?

Reply:
Each group is represented as 1 deployment in Apollo, so this will throttle per deployment object